### PR TITLE
dns: add individual pod endpoints cross cluster also

### PR DESCRIPTION
**Pull request by ramaraochavali**
With this https://github.com/istio/istio/pull/31067 we are skipping cross cluster endpoints. However for individual pod endpoints for stateful sets we do not need load balancing and we need to add it even for cross cluster
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/name_table.go [hunk 1]
```diff
@@ -78,19 +78,7 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
...
-						// endpoints, cross network routing would break, as we do passthrough LB and
-						// don't go through the network gateway. While we could, hypothetically, send
-						// "network-local" endpoints, this would still make enabling DNS give vastly
-						// different load balancing than without, so its probably best to filter.
-						// This ends up matching the behavior of Kubernetes DNS.
-						continue
-					}
-					// TODO: should we skip the node's own IP like we do in listener?
-					addressList = append(addressList, instance.Endpoint.Address)
+					// Add individual addresses even for cross cluster.
```
Comment (stevenctl) [inline pilot/pkg/networking/core/v1alpha3/name_table.go:81]: Can we add some tests for this? And does it make sense to filter down to `network-local` as mentioned in the comment?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/name_table.go:81]: We already have unit tests for individual addresses. We do want to add integration tests with stateful sets in future.

> And does it make sense to filter down to network-local as mentioned in the

Yes. We can do that as follow-up. @howardjohn may have better idea as he added that comment.

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/name_table.go:81]: Oh you mean - individual pod endpoints also filter by network local?

Comment (stevenctl) [reply pilot/pkg/networking/core/v1alpha3/name_table.go:81]: Yeah

Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/name_table.go:81]: I suppose its not directly related to this PR, but do need to handle cross-cluster conflicts better here?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/name_table.go:81]: Yes. Agree. We can think about it and fix in follow-up

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/name_table.go:81]: Ok. Added that. Please check. I can add tests later along with integration test.

**Review by stevenctl — APPROVED** [review]
LGTM

**Comment by ramaraochavali** [comment]
@howardjohn Wondering if we should the same for all addresses i.e. filter by network instead of cluster so that we will have more endpoints to load balance - so if it reachable we should be able to use it. Any concerns with that? If not I will change it.

**Pull request merged**